### PR TITLE
#1552 added bonus points for invitation

### PR DESCRIPTION
**Pull request by krzyk**
#1552
* inviter will now get +32 points for inviting
* updated test for friend invitation
* changed user.json('type') and user.json('name') to correct user methods - now it will work correctly with MkUser
* added new type of message, to be able to award just points without payment

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1638](https://github.com/zerocracy/farm/issues/1638) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1638](https://github.com/zerocracy/farm/issues/1638) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2351359462); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy [hunk 1]
```diff
@@ -120,10 +118,17 @@ def exec(Project pmo, XML xml) {
...
   claim.copy()
-    .type('Make payment')
+    .type('Add award points')
+    .param('login', author)
+    .param('job', 'none')
+    .param('minutes', -new Policy().get('1.price', 128))
+    .param('reason', new Par('Invited @%s').say(login))
+    .postTo(new ClaimsOf(farm))
+  claim.copy()
+    .type('Add award points')
```
Comment (paulodamaso) [inline src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @krzyk I think that we should only give the resume examination bonus if the resume was assigned to the user, and not when we invite a friend directly. Also, the examination bonus claim reason should be something like `@%s resume examination`.

File: src/test/resources/com/zerocracy/bundles/examiner_can_invite/_after.groovy [hunk 2]
```diff
@@ -31,8 +32,13 @@ def exec(Project project, XML xml) {
     people.hasMentor(friend),
     Matchers.is(true)
   )
+  String user = 'user'
   MatcherAssert.assertThat(
```
Comment (paulodamaso) [inline src/test/resources/com/zerocracy/bundles/examiner_can_invite/_after.groovy]: @krzyk we should also check if the messages were sent correctly.

File: src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy (hunk 1 above)
Comment (krzyk) [reply src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @paulodamaso we can invite users only if they have resume under examination (see line 72 in that file)

**Comment by krzyk** [comment]
@paulodamaso updated

File: src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy (hunk 1 above)
Comment (paulodamaso) [reply src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @krzyk But then it does not make much sense. What is the purpose of the ìnvite friend` flow we actually does not invite people to zerocracy, but just ask them to fill some form like everyone else, and there is a chance that this user resume goes to another person to examine, which brokes the invitation process. Don't you agree with that?

Comment (krzyk) [reply src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @paulodamaso Maybe, but it is how it works right now, I didn't change that part, I just added +32 points for invitation.

Comment (paulodamaso) [reply src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @krzyk Just for keeping our peace of mind, would you kindly left a todo about this then?

Comment (krzyk) [reply src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @paulodamaso But that's more like a separate issue (notice that in current todo there was no mention of this flow change).

Comment (paulodamaso) [reply src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @krzyk You are right, as the todo text does not state anything, let's just left it as is.

**Comment by 0crat** [comment]
Job was finished in 24 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1638](https://github.com/zerocracy/farm/issues/1638) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**